core/org.wso2.carbon.core.services/src/main/java/org/wso2/carbon/core/services/fileupload/FileUploadService.java

```diff
@@ -79,15 +83,42 @@
         }
     }
 
+    private void verifyCanonicalDestination(String extraFileLocation, File dirs, String fileName) throws IOException {
+
+        String canonicalDestinationDirPath = dirs.getCanonicalPath();
+        File destinationFile = new File(extraFileLocation, fileName);
+        String canonicalDestinationFile = destinationFile.getCanonicalPath();
+
+        if (!canonicalDestinationFile.startsWith(canonicalDestinationDirPath + File.separator)) {
+            throw new AxisFault("Entry is outside of the target dir: " + fileName);
```

Comment:
why AxisFault exception?

Reply:
we can improve the error message as well.

Reply:
The uploadFiles method is thrown AxisFault exception. So we can throw a AxisFault exception here as well.

Reply:
Fixes the Error message.

Reply:
let's change the throw expectation to `AxisFault`